src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -174,4 +162,26 @@ def validate(self, attrs):
         if not get_current_value("isInternal") and not get_current_value("author"):
             raise ValidationError({"author": "author required for public integrations"})
 
+        # validate globally unique name for public integrations, and org-level uniqueness for internal integrations
+        current_name = get_current_value("name")
+        current_is_internal = get_current_value("isInternal")
+        if current_name:
```

Comment:
@nolachen when would `current_name` be falsy?

Reply:
@scefali Maybe if somehow there was an empty string passed in for the name? I have this because the original function `validate_name` also had a falsy check for the name

Reply:
@nolachen I think that's different since `validate_name` is checking the name that is passed into the API while `get_current_value` will fall back to the value of the existing model if it isn't passed.

Reply:
I think this check should only check if the name is passed in the API so don't use `get_current_value` on the name